lib/improver/standardise.py

```diff
@@ -0,0 +1,446 @@
+        except ValueError:
+            # raised by np.allclose if "end" index overshoots edge of grid
+            # domain - slicing does not raise IndexError
+            return False
+
+    return True
+
+
+class StandardiseGridAndMetadata(BasePlugin):
+
+    """Plugin to regrid cube data and standardise metadata"""
+
+    REGRID_REQUIRES_LANDMASK = {"bilinear": False,
+                                "nearest": False,
+                                "nearest-with-mask": True}
+
+    def __init__(self, regrid_mode='bilinear', extrapolation_mode='nanmask',
+                 landmask=None, landmask_vicinity=25000):
+        """
+        Initialise regridding parameters
+
+        Args:
+            regrid_mode (str):
+                Mode of interpolation in regridding.
+            extrapolation_mode (str):
+                Mode to fill regions outside the domain in regridding.
+            landmask (iris.cube.Cube or None):
+                Land-sea mask ("land_binary_mask") on the input cube grid.
+                Required for "nearest-with-mask" regridding option.
+            landmask_vicinity (float):
+                Radius of vicinity to search for a coastline, in metres
+
+        Raises:
+            ValueError: If a landmask is required but not passed in
+        """
+        if landmask is None and "nearest-with-mask" in regrid_mode:
+            msg = ("An argument has been specified that requires an input "
+                   "landmask cube but none has been provided")
+            raise ValueError(msg)
+        self.regrid_mode = regrid_mode
+        self.extrapolation_mode = extrapolation_mode
+        self.landmask_source_grid = landmask
+        self.landmask_vicinity = (
+            None if landmask is None else landmask_vicinity)
+        self.landmask_name = "land_binary_mask"
+
+        # set attributes to inherit from the target grid
+        self.grid_attributes = [
+            'mosg__grid_version', 'mosg__grid_domain', 'mosg__grid_type']
+
+    def _adjust_landsea(self, cube, target_grid):
+        """
+        Adjust regridded data using differences between the target landmask
+        and that obtained by regridding the source grid landmask, to ensure
+        that the "land" or "sea" nature of the points in the regridded cube
+        matches that of the target grid.
+
+        Returns:
+            iris.cube.Cube: Adjusted cube
+        """
+        if self.landmask_name not in self.landmask_source_grid.name():
+            msg = ("Expected {} in input_landmask cube but found {}".format(
+                self.landmask_name, repr(self.landmask_source_grid)))
+            warnings.warn(msg)
+
+        if self.landmask_name not in target_grid.name():
+            msg = ("Expected {} in target_grid cube but found {}".format(
+                self.landmask_name, repr(target_grid)))
+            warnings.warn(msg)
+
+        plugin = AdjustLandSeaPoints(vicinity_radius=self.landmask_vicinity)
```

Comment:
I don't know, but this would change the logic of what this code has been doing (as currently a different interpolation method for the initial regridding does not propagate to the land-sea mask adjustment).  This might be worth discussing with @bayliffe / @fionaRust and opening follow-on work if necessary.

Reply:
In fact there is actually no need for `AdjustLandSeaPoints` to have this `__init__` argument, as it is never used.  But this may not be a simple question to answer (use or remove), so I'd say it's something to do as a separate change.